product/scanning.py

```diff
@@ -116,40 +116,172 @@ def _retrieve_cached_scan_result(
     ):
         return self.retrieve_scan_result(container_image=container_image, component=component)
 
-    def upload_image(
-            self,
-            container_image: ContainerImage,
-            component: Component,
-            wait_for_result: bool=False
-        ):
-        metadata = self._metadata(container_image=container_image, component=component)
+    def process_image(
+        self,
+        container_image: ContainerImage,
+        component: Component,
+        processing_mode: ProcessingMode,
+        wait_for_result: bool=False,
```

Comment:
tbh, I think this argument was never actually used from callers. so we might just as well remove it
buuuuut: it seems to me that it is now silently ignored, is it not?